tla/consensus/ccfraft.tla

```diff
@@ -766,30 +775,31 @@ ChangeConfigurationInt(i, newConfiguration) ==
     /\ leadershipState[i] = Leader
     \* Configuration is not equal to the previous configuration.
     /\ newConfiguration /= MaxConfiguration(i)
-    \* CCF's integrity demands that a previously removed server cannot rejoin the network,
-    \* i.e., be re-added to a new configuration.  Instead, the node has to rejoin with a
-    \* "fresh" identity (compare sec 6.2, page 8, https://arxiv.org/abs/2310.11559).
-    /\ \A s \in newConfiguration: s \notin removedFromConfiguration
-    \* See raft.h:2401, nodes are only sent future entries initially, they will NACK if necessary.
-    \* This is because they are expected to start from a fairly recent snapshot, not from scratch.
-    \* Note that the sentIndex is set to the log entry *before* the reconfiguration was added
-    \* This is to allow the send AE action to send an initial heartbeat which matches the implementation
-    /\ LET
-        addedNodes == newConfiguration \ CurrentConfiguration(i)
```

Comment:
This was not right, the diffing needs to happen using using the latest/MaxConfiguration, if the CurrentConfiguration is used, intermediary configurations may be present that already have the nodes.

Reply:
Separated into #6047.